src/m365/spo/commands/file/file-list.ts

```diff
@@ -70,46 +81,113 @@ class SpoFileListCommand extends SpoCommand {
-    let requestUrl = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(folderUrl)}')?$expand=${expandParameters}`;
-    if (args.options.output !== 'json') {
-      requestUrl += '&$select=Files/UniqueId,Files/Name,Files/ServerRelativeUrl';
+  private async getFiles(folderUrl: string, args: CommandArgs, index: number, files: FilePropertiesCollection = { value: [] }): Promise<FilePropertiesCollection> {
+    const requestUrl = `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(folderUrl)}')/Files`;
+
+    const fieldsProperties = this.formatSelectProperties(args.options.fields, args.options.output);
+
+    const queryParams = [`$skip=${index}`, `$top=${SpoFileListCommand.thresholdLimit}`];
+
+    if (fieldsProperties.expandProperties.length > 0) {
+      queryParams.push(`$expand=${fieldsProperties.expandProperties.join(',')}`);
+    }
+
+    if (fieldsProperties.selectProperties.length > 0) {
+      queryParams.push(`$select=${fieldsProperties.selectProperties.join(',')}`);
+    }
+
+    if (args.options.filter) {
+      queryParams.push(`$filter=${args.options.filter}`);
     }
-    const requestOptions: any = {
-      url: requestUrl,
+
+    const requestOptions: CliRequestOptions = {
+      url: `${requestUrl}?${queryParams.join('&')}`,
       method: 'GET',
       headers: {
         'accept': 'application/json;odata=nometadata'
       },
       responseType: 'json'
     };
 
-    return request
-      .get<FileFolderCollection>(requestOptions)
-      .then((filesAndFoldersResult: FileFolderCollection) => {
-        filesAndFoldersResult.Files.forEach((file: FileProperties) => files.value.push(file));
-        // If the request is --recursive, call this method for other folders.
-        if (args.options.recursive &&
-          filesAndFoldersResult.Folders !== undefined &&
-          filesAndFoldersResult.Folders.length !== 0) {
-          return Promise.all(filesAndFoldersResult.Folders.map((folder: { ServerRelativeUrl: string; }) => this.getFiles(folder.ServerRelativeUrl, args, files)));
-        }
-        else {
-          return;
+    const filesAndFoldersResult = await request.get<{ value: FileProperties[] }>(requestOptions);
+    filesAndFoldersResult.value.forEach((file: FileProperties) => files.value.push(file));
+
+    if (filesAndFoldersResult.value.length === SpoFileListCommand.thresholdLimit) {
+      await this.getFiles(folderUrl, args, index + SpoFileListCommand.thresholdLimit, files);
+    }
+
+    return files;
+  }
+
+  private async getSubfoldersPages(folderUrl: string, args: CommandArgs, index: number, folders: string[] = []): Promise<string[]> {
+    const requestOptions: CliRequestOptions = {
+      url: `${args.options.webUrl}/_api/web/GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(folderUrl)}')/Folders?$skip=${index}&$top=${SpoFileListCommand.thresholdLimit}`,
```

Comment:
Using the setup `GetFolderByServerRelativeUrl('${formatting.encodeQueryParameter(folderUrl)` you might run into path length issues when the folderURL gets extremely long with deep paths. 

Let's change this to the following setup:

```typescript
GetFolderByServerRelativeUrl(@folderUrl)?@folderUrl=${formatting.encodeQueryParameter(folderUrl)}
```